### PR TITLE
ED-5369 Edit CPT code details

### PR DESCRIPTION
**Pull request by sheikrxnt**
Functionality changes and design changes to edit and delete CPT codes

### CODE REVIEW

File: src/table/actionHeaderFactory.js [hunk 1]
```diff
@@ -69,6 +69,17 @@ function actionFactory(action, actionConfiguration, schema) {
...
     };
+  } else if (action === "inheritedAction") {
+    return (rowIndex, formData, inheritedActionName, onChange) => {
+      if (inheritedActionName === "delete") {
+        let newFormData = formData.slice(0);
+        newFormData.splice(rowIndex, 1);
+        onChange(newFormData);
+      } else {
+        //Edit
+        window.handleCptEditPopUp(rowIndex);
```
Comment (sneu012) [inline src/table/actionHeaderFactory.js:80]: If we go this route, we will need to check if the function exists before we call it.

Comment (sheikrxnt) [reply src/table/actionHeaderFactory.js]: Added the check for the function exists

File: src/table/actionHeaderFactory.js [hunk 2]
```diff
@@ -86,34 +99,111 @@ function actionColumnFrom(
...
+    rowIndex,
+    formData,
+    onChange,
+    inheritedAction
+  ) => {
+    const handleOutsideClick = e => {
+      inheritActionAmmo = false;
+      selectedRow = false;
+    };
+    document.addEventListener("click", handleOutsideClick, false);
```
Comment (sneu012) [inline src/table/actionHeaderFactory.js:115]: When is this event listener getting removed?

File: src/table/actionHeaderFactory.js [hunk 3]
```diff
@@ -86,34 +99,111 @@ function actionColumnFrom(
...
         {text}
+        {rowIndex === selectedRow && inheritActionAmmo}
       </span>
     ),
-    editable: false,
+    editable: false
   };
 }
 
+const inheritedActionClick = (
```
Comment (sneu012) [inline src/table/actionHeaderFactory.js:159]: Can you explain what this is?

Comment (sheikrxnt) [reply src/table/actionHeaderFactory.js]: Updated the naming's

File: src/table/actionHeaderFactory.js (hunk 2 above)
Comment (sheikrxnt) [reply src/table/actionHeaderFactory.js:116]: Added remove event listener

**Review by sneu012 — APPROVED** [review]
Lets plan to review the changes and create tech debts to address them

**Pull request merged**